server/stream.go

```diff
@@ -4932,31 +4932,37 @@ func (mset *stream) processDirectGetLastBySubjectRequest(_ *subscription, c *cli
 	if len(reply) == 0 {
 		return
 	}
-	var minLastSeq uint64
 	hdr, msg := c.msgParts(rmsg)
 	if errorOnRequiredApiLevel(hdr) {
 		hdr := []byte("NATS/1.0 412 Required Api Level\r\n\r\n")
 		mset.outq.send(newJSPubMsg(reply, _EMPTY_, _EMPTY_, hdr, nil, nil, 0))
 		return
 	}
 
-	// This version expects no payload, unless min last seq is specified.
-	if len(msg) != 0 {
-		var req JSApiMsgGetRequest
+	var req JSApiMsgGetRequest
+	if len(msg) > 0 {
 		err := json.Unmarshal(msg, &req)
 		if err != nil {
 			hdr := []byte("NATS/1.0 408 Malformed Request\r\n\r\n")
 			mset.outq.send(newJSPubMsg(reply, _EMPTY_, _EMPTY_, hdr, nil, nil, 0))
 			return
 		}
+	}
 
-		if req.MinLastSeq == 0 {
-			hdr := []byte("NATS/1.0 408 Bad Request\r\n\r\n")
-			mset.outq.send(newJSPubMsg(reply, _EMPTY_, _EMPTY_, hdr, nil, nil, 0))
-			return
-		}
-		minLastSeq = req.MinLastSeq
+	// Error if request fields are provided that don't make sense for this endpoint.
```

Comment:
Should't we also return an error if `key != req.LastFor`?

Reply:
We could do, currently we just overwrite it.

Reply:
Not feeling strongly about this, mostly I was curious